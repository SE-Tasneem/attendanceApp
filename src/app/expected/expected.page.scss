.input-row {
  display: flex;
  align-items: center;
  padding: 0;
}

.input-field {
  flex: 1;
  margin-right: 0.5rem;
}

.add-button {
  margin-left: 0;
}
.user-row {
  display: flex;
  align-items: center;
  border: 1px solid #ccc;
  border-radius: 6px;
  margin: 2px 0.5rem 1rem 0.5rem;
  padding: 0.1rem;
  background-color: #dfe0e185;
}

.icon-col {
  padding-right: 0.1rem; /* Adjust this value to move the icon closer/further */
}

.user-icon {
  font-size: 2em;
}
.close-icon {
  font-size: 2.3em;
}
.name-col {
  flex: 1;
}

.button-col {
  text-align: right;
  padding: 0px;
}
.icon-button {
  --padding-start: 0;
  --padding-end: 5px;
  --padding-top: 0;
  --padding-bottom: 0;
  --border-radius: 0;
}
button {
  padding: 0px;
}
