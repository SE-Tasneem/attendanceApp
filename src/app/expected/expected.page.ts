import { Component } from '@angular/core';
import { UserService } from '../user.service';

@Component({
  selector: 'app-expected',
  templateUrl: 'expected.page.html',
  styleUrls: ['expected.page.scss']
})
export class ExpectedPage {
  users: any[] = [];
  newUser: any = {}
  constructor(private userService: UserService) { }

  ngOnInit() {
    this.fetchUsers();
  }

  async fetchUsers() {
    try {
      this.users = await this.userService.getUsers();
    } catch (error) {
      console.error('Error fetching users', error);
    }
  }

  async addUser() {
    if (this.newUser.name.trim().length > 0) {
      try {
        const addedUser = await this.userService.addUser(this.newUser);
        this.fetchUsers()
        this.newUser = {}; // Reset newUser object
      } catch (error) {
        console.error('Error adding user', error);
      }
    }
  }
  async removeUser(user: any) {
    try {
      await this.userService.removeUser(user.id);
      this.users = this.users.filter(u => u.id !== user.id);
      console.log('User removed:', user);
    } catch (error) {
      console.error('Error removing user', error);
    }
  }
}
