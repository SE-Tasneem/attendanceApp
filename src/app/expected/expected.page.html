<ion-header [translucent]="true">
  <ion-toolbar color="primary">
    <ion-title>
      الأشخاص
    </ion-title>
  </ion-toolbar>
</ion-header>

<ion-content [fullscreen]="true">
  <ion-card>
    <ion-card-header>
      <ion-card-title>إضافة شخص</ion-card-title>
    </ion-card-header>
    <ion-card-content>
      <ion-item class="input-row">
        <ion-input [(ngModel)]="newUser.name" placeholder="ادخل الاسم" class="input-field"></ion-input>
        <ion-button (click)="addUser()" class="add-button">
          <ion-icon name="add-outline" aria-hidden="true"></ion-icon>
        </ion-button>
      </ion-item>
    </ion-card-content>
  </ion-card>

  <ion-list>
    <ion-list-header>
      <ion-card-title>قائمة الأشخاص</ion-card-title>
    </ion-list-header>
    <!-- <ion-item > -->
      <ion-row *ngFor="let user of users" class="user-row" align-items-center>
        <ion-col size="auto" class="icon-col">
          <ion-icon name="person-circle" class="user-icon"></ion-icon>
        </ion-col>
        <ion-col class="name-col">
          <ion-label>{{ user.name }}</ion-label>
        </ion-col>
        <ion-col size="auto" class="button-col">
          <ion-button fill="clear" (click)="removeUser(user)" class="icon-button">
            <ion-icon name="close-circle" color="danger" class="close-icon"></ion-icon>
          </ion-button>
        </ion-col>
      </ion-row>

    <!-- </ion-item> -->
  </ion-list>

  <!-- <app-explore-container name="Expected page"></app-explore-container> -->
</ion-content>
